zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/error/ErrorServiceImpl.java

```diff
@@ -125,8 +136,13 @@ private Message createMessage(String key, Object... parameters) {
             messageParameters = validateParameters(message, key, parameters);
             text = String.format(message.getText(), messageParameters);
         }
-
-        return new BasicMessage(key, message.getType(), message.getNumber(), text);
+        if (message.getComponent() == null) {
+            StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
+            String className = stackTrace[3].getClassName();
```

Comment:
Pure curiosity - is this a common technique, index 3? 

I'm still learning spring 😀

Reply:
@dkelosky Magic numbers are not a good technique. I will address it by using a constant with a good name.

The 3 is a number of stack trace element of the code that called the `ErrorServiceImpl.createApiMessage()` which called the `ErrorServiceImpl.createMessage()`. The number is dependent on the structure of the ErrorServiceImpl and refactoring can break it.  There is a test that verifies that the information about the class is correct.

This picture shows the structure of the call stack:

![image](https://user-images.githubusercontent.com/7644758/67550085-f511ea80-f705-11e9-95e0-90668518080c.png)